lib/features/login/data/network/config/authorization_interceptors.dart

```diff
@@ -1,32 +1,123 @@
+    _authenticationType = newAuthenticationType;
+  }
+
+  void setOIDCConfiguration(OIDCConfiguration? newConfig) {
+    _configOIDC = newConfig;
+  }
 
   void changeAuthorization(String? userName, String? password) {
     _authorization = base64Encode(utf8.encode('$userName:$password'));
   }
 
   void setToken(Token? newToken) {
     _token = newToken;
+    log('AuthorizationInterceptors::setToken(): newToken: $newToken');
+    log('AuthorizationInterceptors::setToken(): EXPIRE_DATE: ${newToken?.expiredTime?.toIso8601String()}');
   }
 
   @override
   void onRequest(RequestOptions options, RequestInterceptorHandler handler) {
-    if (_authorization != null) {
-      options.headers[HttpHeaders.authorizationHeader] = _getAuthorizationAsBasicHeader(_authorization);
-    } else if (_token != null && _token?.isTokenValid() == true) {
-      options.headers[HttpHeaders.authorizationHeader] = _getTokenAsBearerHeader(_token!.token);
+    switch(_authenticationType) {
+      case AuthenticationType.basic:
+        if (_authorization != null) {
+          options.headers[HttpHeaders.authorizationHeader] = _getAuthorizationAsBasicHeader(_authorization);
+        }
+        break;
+      case AuthenticationType.oidc:
+        if (_token != null && _token?.isTokenValid() == true) {
+          options.headers[HttpHeaders.authorizationHeader] = _getTokenAsBearerHeader(_token!.token);
+        }
+        break;
+      case AuthenticationType.none:
+        break;
     }
     super.onRequest(options, handler);
   }
 
+  @override
+  void onError(DioError err, ErrorInterceptorHandler handler) async {
+    final requestOptions = err.requestOptions;
+    if (_isAuthenticationError(err) && _authenticationType == AuthenticationType.oidc) {
+      if (_configOIDC != null && _token != null) {
+        final newToken = await _oidcHttpClient.refreshingTokensOIDC(
+            _configOIDC!.clientId,
+            _configOIDC!.redirectUrl,
+            _configOIDC!.discoveryUrl,
+            _configOIDC!.scopes,
+            _token!.refreshToken);
+
+        await Future.wait([
+          _tokenOidcCacheManager.persistOneTokenOidc(newToken),
+          _accountCacheManager.setSelectedAccount(Account(
+              newToken.tokenId.hashCode.toString(),
+              AuthenticationType.oidc,
+              isSelected: true)),
+        ]);
+
+        setToken(newToken.toToken());
+
+        requestOptions.headers[HttpHeaders.authorizationHeader] =
+            _getTokenAsBearerHeader(newToken.token);
+      } else {
+        requestOptions.headers[HttpHeaders.authorizationHeader] =
+            _getTokenAsBearerHeader(_token!.token);
+      }
+
+      final response = await _dio.fetch(requestOptions);
+      return handler.resolve(response);
+    } else {
+      super.onError(err, handler);
+    }
+  }
+
+  bool _isAuthenticationError(DioError dioError) {
+    if (dioError.type == DioErrorType.response &&
```

Comment:
`response` don't have a dedicated error for `expired`?

Reply:
It is a necessary condition, when it cannot return data from a `token expire`. I have checked `token is expire` in this method.

Reply:
IMO, it should be more safety for us when the server says that we `expired`.

Reply:
if ((_isTokenExpired() || err.response?.statusCode == 401) 
   && _authenticationType == AuthenticationType.oidc) {
 
}